src/k2/module/k23si/Module.cpp

```diff
@@ -840,16 +834,9 @@ K23SIPartitionModule::handleWrite(dto::K23SIWriteRequest&& request, dto::K23SI_M
         auto& rec = versions[0];
         auto& rqmtr = request.mtr;
     
-        // Clean up if req and WI are in the same transaction
```

Comment:
it is interesting that this has to be removed in order for partial updates to work correctly (p-update over WI from same txn scenario). We should probably have a test case for that if we don't already

Reply:
I checked and the existing tests were all partial update over write intent, so I added one that is partial update over committed version.